lib/parse.js

```diff
@@ -349,6 +349,9 @@ function tokenizer($TEXT, filename, html5_comments, shebang) {
             return is_alphanumeric_char(code);
         });
         if (prefix) num = prefix + num;
+        if (RE_OCT_NUMBER.test(num) && next_token.has_directive("use strict")) {
```

Comment:
It would be slightly better to have parse_js_number inside the tokenizer so it can access parse_error (although talking about 1 less check that means almost nothing on cpu). But this would create a slightly complex (but still pretty small) merge conflict.